softlayer/vm/softlayer_vm.go

```diff
@@ -279,3 +290,61 @@ func (vm SoftLayerVM) getRootPassword(virtualGuest datatypes.SoftLayer_Virtual_G
 
 	return ""
 }
+
+func (vm SoftLayerVM) postCheckActiveTransactionsForDeleteVM(softLayerClient sl.Client, virtualGuestId int, timeout, pollingInterval time.Duration) error {
+	virtualGuestService, err := softLayerClient.GetSoftLayer_Virtual_Guest_Service()
+	if err != nil {
+		return bosherr.WrapError(err, "Creating VirtualGuestService from SoftLayer client")
+	}
+
+	totalTime := time.Duration(0)
+	for totalTime < timeout {
+		activeTransactions, err := virtualGuestService.GetActiveTransactions(virtualGuestId)
+		if err != nil {
+			return bosherr.WrapError(err, "Getting active transactions from SoftLayer client")
+		}
+
+		if len(activeTransactions) > 0 {
+			vm.logger.Info(deleteVMLogTag, "Delete VM transaction started", nil)
+			break
+		}
+
+		totalTime += pollingInterval
+		time.Sleep(pollingInterval)
+	}
+
+	if totalTime >= timeout {
+		return bosherr.WrapError(err, "Waiting for DeleteVM transaction to start TIME OUT!")
+	}
+
+	totalTime = time.Duration(0)
+	for totalTime < timeout {
+		vm1, err := virtualGuestService.GetObject(virtualGuestId)
+		if err != nil || vm1.Id == 0 {
+			vm.logger.Info(deleteVMLogTag, "VM doesn't exist. Delete done", nil)
+			break
+		}
+
+		activeTransaction, err := virtualGuestService.GetActiveTransaction(virtualGuestId)
+		if err != nil {
+			return bosherr.WrapError(err, "Getting active transactions from SoftLayer client")
+		}
+
+		averageTransactionDuration, _ := strconv.ParseFloat(activeTransaction.TransactionStatus.AverageDuration, 32)
```

Comment:
Why are you ignoring the `error` of `strconv.ParseFloat`. What if SL sends garbage and the ParseFloat fails.. You need to check `if err != nil` and either recover (e.g., retry or something like that) or return and fail.